pkg/runtime.v2/core/trainingruntime.go

```diff
@@ -106,26 +103,23 @@ func (r *TrainingRuntime) buildObjects(
 		runtime.WithMLPolicy(mlPolicy),
 		runtime.WithPodGroupPolicy(podGroupPolicy),
 	}
-	for idx, rJob := range jobSetTemplateSpec.Spec.ReplicatedJobs {
-		replicas := jobSetTemplateSpec.Spec.ReplicatedJobs[idx].Replicas * ptr.Deref(rJob.Template.Spec.Completions, 1)
-		opts = append(opts, runtime.WithPodSpecReplicas(rJob.Name, replicas, rJob.Template.Spec.Template.Spec))
+	for _, rJob := range jobSetTemplateSpec.Spec.ReplicatedJobs {
```

Comment:
Actually, we won't see panic since TrainJob can't be empty and spec can't be nil: https://github.com/kubeflow/training-operator/blob/master/pkg/apis/kubeflow.org/v2alpha1/trainjob_types.go#L44
We execute this API as part of this script and TrainJob can't be nil there: https://github.com/kubeflow/training-operator/blob/e5165cbff046dba921a1f372293b73ad52888ae6/pkg/controller.v2/trainjob_controller.go#L81